Ginger/GingerCoreNET/RunLib/DynamicExecutionLib/DynamicExecutionManager.cs

```diff
@@ -948,48 +997,113 @@ public static void CreateUpdateRunSetFromJSON(RunsetExecutor runsetExecutor, Gin
+                        }
+                        if (runsetOperationConfigMail.BodyTextContent != null)
+                        {
+                            mailOperation.Bodytext = runsetOperationConfigMail.BodyTextContent;
+                        }
+
+                        //report attachments
+                        EmailAttachment reportAttachment = mailOperation.EmailAttachments.Where(x => x.AttachmentType == EmailAttachment.eAttachmentType.Report).FirstOrDefault();
                         if (runsetOperationConfigMail.IncludeAttachmentReport != null)
                         {
-                            if (runsetOperationConfigMail.IncludeAttachmentReport == true)
+                            if (runsetOperationConfigMail.IncludeAttachmentReport == true && reportAttachment == null)
                             {
-                                if (mailOperation.EmailAttachments.Count == 0)
-                                {
-                                    EmailHtmlReportAttachment reportAttachment = new EmailHtmlReportAttachment();
-                                    reportAttachment.AttachmentType = EmailAttachment.eAttachmentType.Report;
-                                    reportAttachment.ZipIt = true;
-                                    mailOperation.EmailAttachments.Add(reportAttachment);
-                                }
+                                //defualt settings for report attachment
+                                reportAttachment = new EmailHtmlReportAttachment();
+                                reportAttachment.AttachmentType = EmailAttachment.eAttachmentType.Report;
+                                ((EmailHtmlReportAttachment)reportAttachment).SelectedHTMLReportTemplateID = 100;//default report template
+                                reportAttachment.ZipIt = true;
+                                mailOperation.EmailAttachments.Add(reportAttachment);
+                            }
+                            else if (runsetOperationConfigMail.IncludeAttachmentReport == false && reportAttachment != null)
+                            {
+                                mailOperation.EmailAttachments.Remove(reportAttachment);
+                                reportAttachment = null;
+                            }
+                        }
+                        if (reportAttachment != null && runsetOperationConfigMail.AttachmentReportTemplateID != null)
+                        {
+                            ((EmailHtmlReportAttachment)reportAttachment).SelectedHTMLReportTemplateID = (int)runsetOperationConfigMail.AttachmentReportTemplateID;
+                        }
+                        if (reportAttachment != null && runsetOperationConfigMail.AttachmentReportAttachType != null)
+                        {
+                            switch (runsetOperationConfigMail.AttachmentReportAttachType)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5700269)